tests/unit_tests/test_event_service.py

```diff
@@ -37,6 +39,22 @@ def service(api_accessor: ApiAccessor):
     ]
 
     result = await service.execute_batch_update(42, queue)
-    assert result == json.loads(content.decode('utf-8'))['updated_events']
 
-    service.api_accessor.api_post.assert_called_once_with("/events/recordMultiple", 42, data=dict(updates=queue))
+    events_data = []
+    for event in json.loads(content)['data']:
+        converted_event = dict(
+            event_id=event['attributes']['eventId'],
+            count=event['attributes']['count']
+        )
+        events_data.append(converted_event)
+
+
+    assert result == events_data
+
+    service.api_accessor.update_events.assert_called_once_with(queue, 42)
+
+    # API is not working
+    service.api_accessor.update_events.coro.return_value = (500, None)
+    result = await service.execute_batch_update(42, queue)
+    assert result is None
+
```

Comment:
This test you can split in 3 smaller

Reply:
Yes but I would have to repeat the first few steps and assert they work before doing the others, which means if the first test was failing the three would too, and if the second test was failing the remaining would aswell. So I don't know if splitting that in multiple tests is actually better.

Reply:
Splitting into multiple tests is better, even if you have to make some configuration. Configuration can be made via fixtures.

Reply:
If test will fail, you'll know what's going on.